service/task.go

```diff
@@ -640,6 +641,73 @@ func (uis *UIServer) taskLogRaw(w http.ResponseWriter, r *http.Request) {
 	uis.render.Stream(w, http.StatusOK, data, "base", "task_log.html")
 }
 
+func (uis *UIServer) taskFileRaw(w http.ResponseWriter, r *http.Request) {
+	projCtx := MustHaveProjectContext(r)
+	if projCtx.Task == nil {
+		uis.LoggedError(w, r, http.StatusNotFound, errors.New("task not found"))
+		return
+	}
+
+	fileName := gimlet.GetVars(r)["file_name"]
+	if fileName == "" {
+		uis.LoggedError(w, r, http.StatusBadRequest, errors.New("file name not specified"))
+		return
+	}
+
+	taskFiles, err := artifact.GetAllArtifacts([]artifact.TaskIDAndExecution{{TaskID: projCtx.Task.Id, Execution: projCtx.Task.Execution}})
+	if err != nil {
+		uis.LoggedError(w, r, http.StatusInternalServerError, errors.Wrapf(err, "unable to find artifacts for %s", projCtx.Task.Id))
+	}
+	taskFiles, err = artifact.StripHiddenFiles(taskFiles, true)
+	if err != nil {
+		uis.LoggedError(w, r, http.StatusInternalServerError, errors.Wrapf(err, "unable to strip hidden files for %s", projCtx.Task.Id))
+	}
+	var tFile *artifact.File
+	for _, taskFile := range taskFiles {
+		if taskFile.Name == fileName {
+			tFile = &taskFile
+			break
+		}
+	}
+	if tFile == nil {
+		uis.LoggedError(w, r, http.StatusNotFound, errors.New("File not found"))
+		return
+	}
+	if !utility.StringMatchesAnyRegex(tFile.ContentType, uis.Settings.Ui.FileStreamingContentTypes) {
+		uis.LoggedError(w, r, http.StatusBadRequest, errors.New(fmt.Sprintf("File content type: %s not supported", tFile.ContentType)))
+		return
+	}
+
+	response, err := http.Get(tFile.Link)
```

Comment:
Won't this support non-S3 links? Is that something we want now?

Reply:
I don't think it will ever reach this point. Since we perform a check on `content_type` above and that value [only gets set for files uploaded](https://github.com/evergreen-ci/evergreen/pull/6998) with s3.put it should never happen for non-s3 links.